lib/asyn-thread.c

```diff
@@ -251,7 +251,7 @@ int init_thread_sync_data(struct thread_data *td,
 
 #ifndef CURL_DISABLE_SOCKETPAIR
   /* create socket pair or pipe */
-  if(wakeup_create(&tsd->sock_pair[0]) < 0) {
+  if(wakeup_create(tsd->sock_pair, FALSE) < 0) {
```

Comment:
Is there a reason why this socketpair is not set nonblocking?

Reply:
I was just looking for a reason for why `wakeup_create()` would need that second argument for non-blocking or not, and if we instead assume always non-blocking things get a little bit simpler. Right?

Reply:
Yes, that would be more simpler. But I'm not sure someday someone may want to create a blocking wakeup_pair. So maybe keep this flexibility might help in the future.

Reply:
I disagree. Providing flexibility today for something "maybe in the future" is a bad idea. You should write exactly and only for this current use case. In the future more things have changed anyway.

Reply:
True, if we can make it simple for the current situation, it's worth doing it. But the current usage of the blocking pair seems like a real thing, they don't handle the `EAGAIN` error and retry. So I think blocking pair is indeed required by those callers.

Reply:
That is a valid reason for using blocking indeed. The question is perhaps is if it matters. It writes a single byte to the freshly created descriptor that then gets closed. Can it actually do EAGAIN in practice?

But yes, I suppose there are systems or theoretical models where it can happen. You are right. Let's do it blocking for this case, as it matches how it works for the current use case when `pipe() is used.

Reply:
So we can leave it unchanged here and move forward?

Reply:
yes!